### PR TITLE
[PR] Linting markdown files and checking for broken internal and external links

### LINKED ISSUE #10: How to Detect and Fix Broken Links?
Since moving to `Nextra` _many_ of the _internal_ links in our docs are broken  ... 💔 😢 
One thing that `Docusaurus` gave us out-of-the-box was checking links. 🦕 📜 
We need to investigate if `Nextra` has the feature to check all links in a page and confirm they work. 🔗 👍 

# Todo

+ [x] Investigate if `Nextra` has a built-in way to check _internal_ links to confirm they aren't `404`. 📥 
+ [x] Same for _external_ links but this probably won't be a thing as it requires making outbound HTTP requests. 🌐 

If there is an _easy_ way to check the _internal_ links please share and _implement_ on this `demo` with instructions. 📝 🙏 
Please add comments for your searches and pages you read along the way so others don't have to waste the same time. ⏳

### PR DESCRIPTION
**Pull request by LuchoTurtle**
closes #10 

This PR adds a few new features:
- uses https://github.com/tcort/markdown-link-check to recursively find **external URL inside `.mdx` files** and checks if they are alive. It additionally checks if the `mailto` e-mail addresses are valid and do exist. This package goes over the data content generated by https://github.com/contentlayerdev/contentlayer to find the links.
- has a script that uses https://github.com/theoludwig/markdownlint-rule-relative-links (and subsequently https://github.com/DavidAnson/markdownlint-cli2) to lint through all the `.mdx` files, applying linting according to default rules. These rules can be found at https://github.com/DavidAnson/markdownlint#rules--aliases. This script **checks for the integrity of internal links**, finding broken anchors between local documents (including sections). 
- both of these processes have been added to CI.
- script is tested.

### CODE REVIEW

File: tutorial.md [hunk 1]
```diff
@@ -2384,28 +2383,10 @@ which is different to what this directory requires.
...
-
+# Star the repo! 
 
 Thanks for learning with us!
 If you find it useful, 
 please give the repo a star! ⭐️
 
-[![HitCount](https://hits.dwyl.com/dwyl/nextra-demo-tutorial.svg?style=flat-square)](https://hits.dwyl.com/dwyl/nextra-demo)
\ No newline at end of file
+[![HitCount](https://hits.dwyl.com/dwyl/nextra-demo.svg?style=flat-square)](https://hits.dwyl.com/dwyl/nextra-demo)
```
Comment (nelsonic) [inline tutorial.md:2392]: Very specifically have a _distinct_ `.svg` URL for the `tutorial.md` 
to count how many people follow through to it as opposed to the main `README.md`
```suggestion
[![HitCount](https://hits.dwyl.com/dwyl/nextra-demo-tutorial.svg?style=flat-square)](https://hits.dwyl.com/dwyl/nextra-demo)
```

File: scripts/link-check.mjs [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import core from "@actions/core"
```
Comment (nelsonic) [inline scripts/link-check.mjs:1]: does this script have any _tests_...? 🧪 💭

Comment (LuchoTurtle) [reply scripts/link-check.mjs:1]: It will, I'm focusing on doing that and finishing the `README`.

**Comment by LuchoTurtle** [comment]
Keep getting this error, darn it.

```
   Jest encountered an unexpected token

    Jest failed to parse a file. This happens e.g. when your code or its dependencies use non-standard JavaScript syntax, or when Jest is not configured to support such syntax.

    Out of the box Jest supports Babel, which will be used to transform your files into valid JS based on your Babel configuration.

    By default "node_modules" folder is ignored by transformers.

    Here's what you can do:
     • If you are trying to use ECMAScript Modules, see https://jestjs.io/docs/ecmascript-modules for how to enable it.
     • If you are trying to use TypeScript, see https://jestjs.io/docs/getting-started#using-typescript
     • To have some of your "node_modules" files transformed, you can specify a custom "transformIgnorePatterns" in your config.
     • If you need a custom transformation specify a "transform" option in your config.
     • If you simply want to mock your non-JS modules (e.g. binary assets) you can stub them out with the "moduleNameMapper" config option.

    You'll find more details and examples of these config options in the docs:
    https://jestjs.io/docs/configuration
    For information about custom transformations, see:
    https://jestjs.io/docs/code-transformation

    Details:

    /Users/lucho/Documents/dwyl/nextra-demo/node_modules/.pnpm/uuid@8.3.2/node_modules/uuid/dist/esm-browser/index.js:1
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,jest){export { default as v1 } from './v1.js';
                                                                                      ^^^^^^

    SyntaxError: Unexpected token 'export'

       9 |
      10 | /**
    > 11 |  * This function checks external links that are present in the markdown files.
         |                                                      ^
      12 |  * It checks if they're alive or dead using `markdown-lint-check`.
      13 |  */
      14 | export function checkMarkdownExternalLinks(markdownBody) {
```

**Comment by LuchoTurtle** [comment]
`jest` really is config hell for ESM :(

**Comment by LuchoTurtle** [comment]
Trying to debug 

```
ReferenceError: You are trying to `import` a file after the Jest environment has been torn down. From tests/unit/link-check.test.ts.
```

for ages now. 

`markdown-link-check` using `async` messes up with Jest. The test effectively passes correctly (as it should) but Jest complains something is trying to be imported after the test is passed.

**Comment by LuchoTurtle** [comment]
@nelsonic this should be ready for review 👌

**Review by nelsonic — APPROVED** [review]
Nice work @LuchoTurtle. 👍 
Thanks. 😊

**Pull request merged**